File: A5/src/exp/VarExp.java
package exp;
import dictionary.*;
import model.MyStmtException;
public class VarExp extends Exp
{
	String id;
	//more stuff to do
	public VarExp(String id)
	{
		this.id = id;
	}
	public int eval(MyIDictionary<String, Integer> tbl, MyIDictionary<Integer, Integer> h) throws MyStmtException
	{
		return tbl.lookup(this.id);
	}
	public String toString()
	{
		return id;
	}
}

File: A7/Expr/VarExp.cs
using A7.Dictionary;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Expr
{
    class VarExp: Exp
    {
        String id;
        public VarExp(String id)
        {
            this.id = id;
        }
        public override int Eval(MyIDictionary<String, int> tbl)
        {
            return tbl.Lookup(this.id);
        }
        public override String ToString()
        {
            return id;
        }
    }
}
